core/services/headreporter/telemetry_reporter.go

```diff
@@ -0,0 +1,65 @@
+package headreporter
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/services/telemetry"
+)
+
+type telemetryReporter struct {
+	endpoints map[uint64]commontypes.MonitoringEndpoint
+}
+
+func NewTelemetryReporter(chainContainer legacyevm.LegacyChainContainer, monitoringEndpointGen telemetry.MonitoringEndpointGenerator) HeadReporter {
+	endpoints := make(map[uint64]commontypes.MonitoringEndpoint)
+	for _, chain := range chainContainer.Slice() {
+		endpoints[chain.ID().Uint64()] = monitoringEndpointGen.GenMonitoringEndpoint("EVM", chain.ID().String(), "", synchronization.HeadReport)
```

Comment:
We could decouple things a little bit here by accepting the slice of chain IDs instead of a container to fetch them from:
```suggestion
func NewTelemetryReporter(monitoringEndpointGen telemetry.MonitoringEndpointGenerator, chainIDs ...*big.Int) HeadReporter {
	endpoints := make(map[uint64]commontypes.MonitoringEndpoint)
	for _, chainID := range chainIDs {
		endpoints[chainID.Uint64()] = monitoringEndpointGen.GenMonitoringEndpoint("EVM", chainID.String(), "", synchronization.HeadReport)
```

Reply:
done